django-stubs/db/models/fields/json.pyi

```diff
@@ -1,34 +1,100 @@
-from typing import Any, Optional
+import json
+from typing import Any, Callable, Iterable, Optional, Tuple, TypeVar, Union, overload
 
 from django.db.models import lookups
 from django.db.models.lookups import PostgresOperatorLookup, Transform
+from typing_extensions import Literal
 
-from . import Field
+from . import Field, _ErrorMessagesToOverride, _ValidatorCallable
 from .mixins import CheckFieldDefaultMixin
 
-class JSONField(CheckFieldDefaultMixin, Field[Any, Any]):
-    empty_strings_allowed: bool = ...
-    description: Any = ...
+_A = TypeVar("_A", bound=Optional[Any])
+
+class JSONField(CheckFieldDefaultMixin, Field[_A, _A]):
     default_error_messages: Any = ...
-    encoder: Any = ...
-    decoder: Any = ...
-    def __init__(
-        self,
-        verbose_name: Optional[Any] = ...,
-        name: Optional[Any] = ...,
-        encoder: Optional[Any] = ...,
-        decoder: Optional[Any] = ...,
-        **kwargs: Any
-    ) -> None: ...
-    def check(self, **kwargs: Any) -> Any: ...
-    def deconstruct(self) -> Any: ...
```

Comment:
do check, deconstruct, and the other deleted methods not exist  on the JSONField

or maybe they're in the parent class?

Reply:
`check` and `deconstruct` exists on `Field`, and theyu are actually better typed there already. No need to redefine them here.

Everything else (like get_internal_type, validate, formfield, etc) are all also defined in `Field`.

This is something that I've done for all fields. I removed what is already declared there and improved `__init__` with the actual field signature